bpf/lib/nodeport.h

```diff
@@ -2067,6 +2092,13 @@ nodeport_extract_dsr_v4(struct __ctx_buff *ctx,
 			*dsr = true;
 			*addr = bpf_ntohl(opt.addr);
 			*port = bpf_ntohs(opt.port);
+
+#ifdef REMOVE_DSR_IP_OPTION
+			int ret = remove_dsr_ip_opt_v4(ctx, ip4, &opt);
+			if (IS_ERR(ret))
+				return ret;
+#endif /* REMOVE_DSR_IP_OPTION */
```

Comment:
Think we want a bit of compile coverage for `REMOVE_DSR_IP_OPTION` in bpf/Makefile. Adding it to some complexity test configs for `bpf_host` / `bpf_xdp` would be cool too.